src/utils/paymentRequest.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+import bcoin from 'bcoin'
+import parse from 'url-parse'
+// $FlowFixMe
+import type { AbcPaymentProtocolInfo } from 'edge-core-js'
+import { toNewFormat } from './addressFormat/addressFormatIndex.js'
+
+// $FlowFixMe
+export const parsePayment = (
+  paymentBuffer: Buffer,
+  network: string,
+  currencyCode: string
+): AbcPaymentProtocolInfo => {
+  const bip70 = bcoin.bip70.PaymentRequest.fromRaw(paymentBuffer)
+  const {
+    paymentUrl = '',
+    memo,
+    merchantData,
+    outputs = []
+  } = bip70.paymentDetails
+  const domain = parse(paymentUrl, true).hostname
+  const abcSpendTarget = []
+  let nativeAmount = 0
+
+  for (const output of outputs) {
+    const jsonObj = output.getJSON(network)
+    nativeAmount += jsonObj.value
+    abcSpendTarget.push({
+      currencyCode: currencyCode,
+      publicAddress: toNewFormat(jsonObj.address, network),
+      nativeAmount: `${jsonObj.value}`
+    })
+  }
+
+  const abcPaymentProtocolInfo = {
+    nativeAmount: `${nativeAmount}`,
```

Comment:
BIP21 specifies amounts in exchangeAmount (ie BTC) not satoshis. Does bcoin for sure return nativeAmounts?

Reply:
I'm pretty sure. I think the test covers that